### PR TITLE
Deprecate IgxDropDownItemBase isSelected && isFocused

### LINKED ISSUE #3612: Depreacte `isSelected` and `isFocused` properties on igx-combo-item
### Description  
The `isSelected` and `isFocused` properties on `IgxComboItemComponent` do not use the proper naming convention and are inconsistent with the other components. The existing properties need to be deprecated and new get/set accessors with the proper names need to be added (`selected` and `focused`, resp.) in order for the component to be consistent.

 * igniteui-angular version: 7.2.x
 * browser: N/A

### Result  
Combo item selected and focus states cannot be input using `selected` / `focused` properties

### Expected result  
Combo item selected and focus states can be input using `selected` / `focused` properties. The `isSelected` and `isFocused` properties will continue to work for the time being (deprecated)

### PR DESCRIPTION
**Pull request by ViktorSlavov**
Closes #3612  

Rename `isSelected` and `isFocused` accessors. Add `selected` and `focused` accessors (with the same functionality)
 
### Additional information (check all that apply):
 - [x] Refactor

### Checklist:
 - [x] All relevant tags have been applied to this PR

### CODE REVIEW

**Comment by bazal4o** [comment]
Do we need to add migrations to that? Or we need to add them once we remove them completely?

**Comment by bkulov** [comment]
The migrations should be added now.

File: projects/igniteui-angular/src/lib/drop-down/drop-down-item.base.ts [hunk 1]
```diff
@@ -56,14 +57,34 @@ export abstract class IgxDropDownItemBase implements DoCheck {
...
+        }
+        this._isSelected = value;
+    }
+
+    /**
+     * @hidden
+     */
+    @Input()
+    @DeprecateProperty(`IgxDropDownItemBase \`isSelected\` property is depracated.\n` +
+        `Use \`selected\` instead.`)
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/drop-down/drop-down-item.base.ts]: You're deprecating `isSelected` and `isFocused` properties and introducing correspondent `selected` and `focused` properties. As far as I can see the getters/setters of those properties have the exact same code. In this case I recommend removing the code from the deprecated properties and get/set the ones instead. By doing that you'll accomplish two things: will ensure that if someone in the future changes the code of the new properties' getter/setter the old ones will continue to work correctly; will avoid duplicate code.

File: projects/igniteui-angular/migrations/common/UpdateChanges.spec.ts [hunk 2]
```diff
@@ -193,6 +193,20 @@ describe('UpdateChanges', () => {
         update3.applyChanges();
         expect(appTree.readContent('test.component.html')).toEqual(
             `<one [replaceMe]="a"> <comp\r\ntag [replaced]="dwdw"> </other> <another oldProp="b" />`);
+
+        inputJson.changes[1] = {
```
[the remark this answers is not in the document]
Comment (ViktorSlavov) [reply projects/igniteui-angular/migrations/common/UpdateChanges.spec.ts]: You're correct, the case I was testing did not need the overwrite explicitly - I've removed it and changed the test string. The point of the test is to make sure that the script does not change all mentions of `change.name`, but only the ones that are used as property bindings.

File: projects/igniteui-angular/src/lib/drop-down/drop-down-item.base.ts [hunk 3]
```diff
@@ -101,6 +119,22 @@ export abstract class IgxDropDownItemBase implements DoCheck {
...
+    }
+
+    /**
+     * @hidden
+     */
+    @HostBinding('class.igx-drop-down__item--focused')
+    @DeprecateProperty(`IgxDropDownItemBase \`isFocused\` property is depracated.\n` +
+        `Use \`focused\` instead.`)
+    get isFocused(): boolean {
+        return (!this.isHeader && !this.disabled) && this._isFocused;
```
[the remark this answers is not in the document]
Comment (ViktorSlavov) [reply projects/igniteui-angular/src/lib/drop-down/drop-down-item.base.ts]: An oversight, thanks for pointing it out!

File: projects/igniteui-angular/migrations/update-7_2_0/index.spec.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+                    focused="true"
+                    [isHeader]="item.thisPropertyIsNOTisSelectedORisFocused"
+                >Selected: {{ item.isSelected }}, Focused {{ item.isFocused }}
+                </igx-drop-down-item>
+            </igx-drop-down>`);
+
+        done();
+    });
+
+    it(`should replace **ONLY** 'isSelected' and 'isFocused'`, done => {
```
[the remark this answers is not in the document]
Comment (ViktorSlavov) [reply projects/igniteui-angular/migrations/update-7_2_0/index.spec.ts]: I've removed the first two tests and left only the most extensive one.

**Pull request merged**